gdal/ogr/ogrpolyhedralsurface.cpp

```diff
@@ -0,0 +1,857 @@
+                                             int nSize,
+                                             OGRwkbVariant eWkbVariant )
+
+{
+    oMP.nGeomCount = 0;
+    OGRwkbByteOrder eByteOrder = wkbXDR;
+    int nDataOffset = 0;
+    OGRErr eErr = importPreambuleOfCollectionFromWkb( pabyData,
+                                                      nSize,
+                                                      nDataOffset,
+                                                      eByteOrder,
+                                                      9,
+                                                      oMP.nGeomCount,
+                                                      eWkbVariant );
+
+    if( eErr != OGRERR_NONE )
+        return eErr;
+
+    /* coverity[tainted_data] */
+    oMP.papoGeoms = (OGRGeometry **) VSI_CALLOC_VERBOSE(sizeof(void*), oMP.nGeomCount);
+    if (oMP.nGeomCount != 0 && oMP.papoGeoms == NULL)
+    {
+        oMP.nGeomCount = 0;
+        return OGRERR_NOT_ENOUGH_MEMORY;
+    }
+
+/* -------------------------------------------------------------------- */
+/*      Get the Geoms.                                                  */
+/* -------------------------------------------------------------------- */
+    for( int iGeom = 0; iGeom < oMP.nGeomCount; iGeom++ )
+    {
+        // Parse the polygons
+        unsigned char* pabySubData = pabyData + nDataOffset;
+        if( nSize < 9 && nSize != -1 )
+            return OGRERR_NOT_ENOUGH_DATA;
+
+        OGRwkbGeometryType eSubGeomType;
+        eErr = OGRReadWKBGeometryType( pabySubData, eWkbVariant, &eSubGeomType );
+        if( eErr != OGRERR_NONE )
+            return eErr;
+
+        OGRGeometry* poSubGeom = NULL;
+        eErr = OGRGeometryFactory::createFromWkb( pabySubData, NULL, &poSubGeom, nSize, eWkbVariant );
+
+        if( eErr != OGRERR_NONE )
+        {
+            oMP.nGeomCount = iGeom;
+            delete poSubGeom;
+            return eErr;
+        }
+
+        oMP.papoGeoms[iGeom] = poSubGeom;
+
+        if (oMP.papoGeoms[iGeom]->Is3D())
+            flags |= OGR_G_3D;
+        if (oMP.papoGeoms[iGeom]->IsMeasured())
+            flags |= OGR_G_MEASURED;
+
+        int nSubGeomWkbSize = oMP.papoGeoms[iGeom]->WkbSize();
+        if( nSize != -1 )
+            nSize -= nSubGeomWkbSize;
+
+        nDataOffset += nSubGeomWkbSize;
+    }
+
+    return OGRERR_NONE;
+}
+
+/************************************************************************/
+/*                            exportToWkb()                             */
+/*      Build a well known binary representation of this object.        */
+/************************************************************************/
+
+OGRErr  OGRPolyhedralSurface::exportToWkb ( OGRwkbByteOrder eByteOrder,
+                                            unsigned char * pabyData,
+                                            OGRwkbVariant eWkbVariant ) const
+
+{
+    if( eWkbVariant == wkbVariantOldOgc &&
```

Comment:
Copy & paste code from OGRGeometryCollection::exportToWkb() that will always evaluate to false. --> remove